advanced/wallets/react-wallet-v2/src/hooks/useSmartAccount.ts

```diff
@@ -1,52 +1,16 @@
 import { SmartAccountLib } from "@/lib/SmartAccountLib";
 import SettingsStore from "@/store/SettingsStore";
-import { Chain, VITALIK_ADDRESS } from "@/utils/SmartAccountUtils";
-import { useCallback, useEffect, useState } from "react";
+import { Chain } from "@/utils/SmartAccountUtils";
+import { useEffect, useState } from "react";
 import { useSnapshot } from "valtio";
 import { Hex } from "viem";
-import { styledToast } from "@/utils/HelperUtil";
-import { TransactionExecutionError } from "viem";
+import { SmartAccount } from "permissionless/accounts";
 
 export default function useSmartAccount(signerPrivateKey: Hex, chain: Chain) {
-    const [loading, setLoading] = useState(false)
     const [client, setClient] = useState<SmartAccountLib>();
-    const [isDeployed, setIsDeployed] = useState(false)
     const [address, setAddress] = useState<Hex>()
     const { smartAccountSponsorshipEnabled } = useSnapshot(SettingsStore.state);
 
-    const execute = useCallback(async (callback: () => void) => {
-      try {
-        setLoading(true)
-        const res = await callback()
-        console.log('result:', res)
-        setLoading(false)
-      }
-      catch (e) {
-        if (e instanceof TransactionExecutionError) {
-          // shorten the error message
-          styledToast(e.cause.message, 'error')
-        } else if (e instanceof Error) {
-          styledToast(e.message, 'error')
-        }
-        console.error(e)
-        setLoading(false)
-      }  
-    }, [setLoading])
-
-    const deploy = useCallback(async () => {
-      if (!client) return
-      execute(client?.deploySmartAccount)
-    }, [client, execute])
-
-    const sendTestTransaction = useCallback(async () => {
-      if (!client) return
-      execute(() => client?.sendTransaction({
```

Comment:
not against removing this but it'd leave the hook without access to the client at all, do you plan on moving this somewhere else / handling it differently ?